README.md

```diff
@@ -1,83 +1,107 @@
 
 ```ruby
-Semian.register(:mysql_master, tickets: 3, timeout: 0.5, error_threshold: 3, error_timeout: 10, success_threshold: 2)
+gem 'semian', require: %w(semian semian/mysql2 semian/redis)
 ```
 
-Then in your child processes, you can use the resource:
+We recommend this pattern of requiring adapters directly from the `Gemfile`.
+This makes ensures Semian adapters is loaded as early as possible, to also
+protect your application during boot. Please see the [adapter configuration
+section](#Configuration) on how to configure adapters.
+
+## Adapters
+
+The following adapters are in Semian and work against the public gems:
+
+* [`semian/mysql2`][mysql-semian-adapter] (~> 0.3.16)
+* [`semian/redis`][redis-semian-adapter] (~> 3.2.1)
+
+### Configuration
+
+When instantiating a resource it now needs to be configured for Semian. This is
+done by passing `semian` as an argument when initializing the client. Examples
+built in adapters:
 
 ```ruby
-Semian[:mysql_master].acquire do
-  # Query the database. If three other workers are querying this resource at the
-  # same time, this block will block for up to 0.5s waiting for another worker
-  # to release a ticket. Otherwise, it'll raise `Semian::TimeoutError`.
-end
+# MySQL2 client
+# In Rails this means having a Semian key in database.yml for each db.
+client = Mysql2::Client.new(host: "localhost", username: "root", semian: {
+  name: "master",
+  tickets: 8, # See the Understanding Semian section on picking these values
+  success_threshold: 2,
+  error_threshold: 3,
+  error_timeout: 10
+})
+
+# Redis client
+client = Redis.new(semian: {
+  name: "inventory",
+  tickets: 4,
+  success_threshold: 2,
+  error_threshold: 4,
+  error_timeout: 20
+})
 ```
 
-If you have a process that doesn't `fork`, you can still use the same namespace
-to control access to a shared resource:
+### Creating an adapter
 
-```ruby
-Semian.register(:mysql_master, timeout: 0.5)
+To create a Semian adapter you must implement the following methods:
 
-Semian[:mysql_master].acquire do
-  # Query the resource
-end
-```
+1. [`include Semian::Adapter`][semian-adapter]. Use the helpers to wrap the
+   resource. This takes care of situations such as monitoring, nested resources,
+   unsupported platforms, creating the Semian resource if it doesn't already
+   exist and so on.
+2. `#semian_identifier`. This is responsible for returning a symbol that
+   represents every unique resource, for example `redis_master` or
+   `mysql_shard_1`. This is usually assembled from a `name` attribute on the
+   Semian configuration hash, but could also be `<host>:<port>`.
+3. `connect`. The name of this method varies. You must override the driver's
+   connect method with one that wraps the connect call with
+   `Semian::Resource#acquire`. You should do this at the lowest possible level.
+4. `query`. Same as `connect` but for queries on the resource.
+
+The best resource is looking at the [already implemented adapters](#adapters).
```

Comment:
You also must define the resource exceptions: `ResourceBusyError` and `CircuitOpenError`. And it's better if they inherits from the adapter base exception.